### ISSUE TITLE
repairing teleport tubes w/ a hammer doesn't restore them to functionality.

### ISSUE THREAD

**Issue by fluxionary** [mt-mods/pipeworks#46]
a player reports:

When a sender teleport tube (tp) is broken, use hammer to repair. Now the tp tube sender is not working anymore. Expected: tp tube sender should tp the items as it did before breaking, doesn't work even after setting the channel name again. Workaround: destroy the receiver tube and then place it again and configure to get the sender tp tube working again.

**Comment by fluxionary** [comment]
> Looks like #36, if this is happens with up to date version then I guess that functionality is now broken.

server is using https://github.com/mt-mods/pipeworks/commit/b6c02ac8deb26bbd73b9bbb94c840e136d224e6e, which is after #37, might be a regression.

**Comment by fluxionary** [comment]
i recently discovered there's a separate issue that causes receive tubes to get delisted, if mod_storage fails to serialize w/ a file-based backend, due to it getting too large for LuaJIT.it's possible it affected this issue as well, i haven't done any thorough testing.

**Comment by TurkeyMcMac** [comment]
The move to mod storage was supposed to avoid the LuaJIT size limit. Each entry is serialized individually by Lua. Could you provide more details of this error?

**Comment by fluxionary** [comment]
> The move to mod storage was supposed to avoid the LuaJIT size limit. Each entry is serialized individually by Lua. Could you provide more details of this error?

possibly i'm wrong, it looks like the file-based metadata is being stored in JSON and not via `minetest.serialize`.

**Comment by fluxionary** [comment]
in any event, for some reason, the "receive" tubes are disappearing from the tube DB "randomly" (which might be after a server restart). i still suspect this is because things aren't being written cleanly when using the file-based backend when the server crashes.

**Comment by TurkeyMcMac** [comment]
The mod storage should be written unless there is an abnormal crash without a backtrace etc.

**Comment by fluxionary** [comment]
> The mod storage should be written unless there is an abnormal crash without a backtrace etc.

yup, that does happen from time to time. moving to sqlite backend should (mostly) make that a non-issue.
